### DISCUSSION TITLE
Does Medley Online support SSH keys for SFTP?

### DISCUSSION THREAD

**Discussion by pamoroso** [Interlisp/medley#1367] [Q&A]
Does Medley Online support SSH keys for authentication with SFTP in addition to passwords?

**Comment by fghalasz** [comment]
No.  A "new" SFTP server is setup on every "Run Medley" within the Docker container (on AWS) for that Medley run.  That new server is set up with a randomly chosen password.

Setting the server up with SSH keys would be a much more difficult issue.  The Web Portal would have to keep a record of each user's SFTP public key and would have to communicate the proper key into the Docker container on every Run Medley.  I considered this but it seemed too complicated for its benefit.  I focused on the web-based Medley file explorer instead figuring people would use that more frequently than they would use SFTP.

Did I get that wrong?

-- Frank

**Reply by fghalasz** [reply]
Actually, on second thought, we could keep the SSH public key in each user's Medley Docker volume that's kept online.   Only thing to figure out is a scheme to get the public key installed on that volume.

I'll open an issue/feature request on this.

-- Frank

**Comment by pamoroso** [comment]
Makes sense, thanks.
